### ISSUE TITLE
buffer overflow detected

### ISSUE THREAD

**Issue by g10rgio** [PoC-Consortium/cg_obup#13]
setting the starting nonce pretty high causes a buffer overflow.

# ./plot64 -k xxxxx -d /data -s 10604000000 -t 12 -n 1000000 -m 10000
Using ORIG core.
Creating plots for nonces 10604000000 to 10605000000 (262 GB) using 2500 MB memory and 12 threads
*** buffer overflow detected ***: ./plot64 terminated
======= Backtrace: =========
/lib/x86_64-linux-gnu/libc.so.6(+0x777e5)[0x7f16fd5727e5]
/lib/x86_64-linux-gnu/libc.so.6(__fortify_fail+0x5c)[0x7f16fd61415c]
/lib/x86_64-linux-gnu/libc.so.6(+0x117160)[0x7f16fd612160]
/lib/x86_64-linux-gnu/libc.so.6(+0x1166c9)[0x7f16fd6116c9]
/lib/x86_64-linux-gnu/libc.so.6(_IO_default_xsputn+0x80)[0x7f16fd5766b0]
/lib/x86_64-linux-gnu/libc.so.6(_IO_vfprintf+0x7bd)[0x7f16fd54892d]
/lib/x86_64-linux-gnu/libc.so.6(__vsprintf_chk+0x84)[0x7f16fd611754]
/lib/x86_64-linux-gnu/libc.so.6(__sprintf_chk+0x7d)[0x7f16fd6116ad]
./plot64[0x401133]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf0)[0x7f16fd51b830]
./plot64[0x4019f9]
======= Memory map: ========
00400000-0040f000 r-xp 00000000 08:01 1050377                            /root/cg_obup/plot64
0060e000-0060f000 r-xp 0000e000 08:01 1050377                            /root/cg_obup/plot64
0060f000-00610000 rwxp 0000f000 08:01 1050377                            /root/cg_obup/plot64
022d3000-022f4000 rwxp 00000000 00:00 0                                  [heap]
7f1660ee4000-7f1660efa000 r-xp 00000000 08:01 1308690                    /lib/x86_64-linux-gnu/libgcc_s.so.1
7f1660efa000-7f16610f9000 ---p 00016000 08:01 1308690                    /lib/x86_64-linux-gnu/libgcc_s.so.1
7f16610f9000-7f16610fa000 rwxp 00015000 08:01 1308690                    /lib/x86_64-linux-gnu/libgcc_s.so.1
7f16610fa000-7f16fd4fb000 rwxp 00000000 00:00 0 
7f16fd4fb000-7f16fd6bb000 r-xp 00000000 08:01 1313106                    /lib/x86_64-linux-gnu/libc-2.23.so
7f16fd6bb000-7f16fd8bb000 ---p 001c0000 08:01 1313106                    /lib/x86_64-linux-gnu/libc-2.23.so
7f16fd8bb000-7f16fd8bf000 r-xp 001c0000 08:01 1313106                    /lib/x86_64-linux-gnu/libc-2.23.so
7f16fd8bf000-7f16fd8c1000 rwxp 001c4000 08:01 1313106                    /lib/x86_64-linux-gnu/libc-2.23.so
7f16fd8c1000-7f16fd8c5000 rwxp 00000000 00:00 0 
7f16fd8c5000-7f16fd8dd000 r-xp 00000000 08:01 1313105                    /lib/x86_64-linux-gnu/libpthread-2.23.so
7f16fd8dd000-7f16fdadc000 ---p 00018000 08:01 1313105                    /lib/x86_64-linux-gnu/libpthread-2.23.so
7f16fdadc000-7f16fdadd000 r-xp 00017000 08:01 1313105                    /lib/x86_64-linux-gnu/libpthread-2.23.so
7f16fdadd000-7f16fdade000 rwxp 00018000 08:01 1313105                    /lib/x86_64-linux-gnu/libpthread-2.23.so
7f16fdade000-7f16fdae2000 rwxp 00000000 00:00 0 
7f16fdae2000-7f16fdb08000 r-xp 00000000 08:01 1313104                    /lib/x86_64-linux-gnu/ld-2.23.so
7f16fdcfc000-7f16fdd00000 rwxp 00000000 00:00 0 
7f16fdd06000-7f16fdd07000 rwxp 00000000 00:00 0 
7f16fdd07000-7f16fdd08000 r-xp 00025000 08:01 1313104                    /lib/x86_64-linux-gnu/ld-2.23.so
7f16fdd08000-7f16fdd09000 rwxp 00026000 08:01 1313104                    /lib/x86_64-linux-gnu/ld-2.23.so
7f16fdd09000-7f16fdd0a000 rwxp 00000000 00:00 0 
7fff3b929000-7fff3b94a000 rwxp 00000000 00:00 0                          [stack]
7fff3b94d000-7fff3b950000 r--p 00000000 00:00 0                          [vvar]
7fff3b950000-7fff3b952000 r-xp 00000000 00:00 0                          [vdso]
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]
Aborted (core dumped)

**Comment by g10rgio** [comment]
I found out what the problem was myself and fixed it.  
On lines 597 and 598 to variables are defined for the filename of the plot (name and finalname).  These are :

char name[100];
char finalname[100];

Because the plot dir that I'm using is actually /data/7fae1645-012f-46a4-bfba-9f244b60898a (the id is the blkid of the hd for reference), then the full dirname + filename was longer than the 100 characters allowed for the variables.

I've changed these variables to a size of 200 and it works now.
